server/business-process-query/get-data.js

```diff
@@ -125,24 +219,105 @@ module.exports = exports = memoize(function (driver) {
 				} else {
 					keyPath = id.slice(bpId.length + 1);
 				}
+
+				// Business process properties
 				meta = exports.businessProcessMetaMap[keyPath];
-				if (meta) {
-					if (meta.type && (meta.type !== 'direct')) return;
-					if (multiItemValue && !meta.multiple) return;
-					if (!meta.validate(record, multiItemValue)) return;
+				if (validateRecord(record, meta, multiItemValue)) {
 					meta.set(initBpDataset(bpId), record, multiItemValue);
+					return;
+				}
+				// Business process nested entities
+				if (exports.businessProcessNestedEntities.some(function (nestedEntityConf) {
```

Comment:
I see this generic map was made purely to gather statusHistory, it's not really needed. We also need to be careful what we do here, as performance and outcome of this scan has strong influence on time of system startup, and on consumed memory of a process.

Gathering `statusHistory` should be done with minimal effort and no trace of it should be left after we resolved _dates map_.

Therefore it definitely should not be mixed into data that we keep at `result.businessProcesses` and `result.steps`, those are entities that are meant to be kept forever in memory and serve very different purpose. Mixing those two different things into same data structures, makes things dirty

`statusHistory` data should be stored  on some other __temporary__ object (we may keep it e.g. on `result._statusHistory`), which we can get rid of easily afterward we have _dates map_ via e.g.:

```javascript
delete result._statusHistory
```

Data at `result._statusHistory` could be stored it form:

```javascript
businessProcesess[businessProcessId] = { status1Id: { stamp: .. , status: .., ..},  status2Id: { .. } }
processingSteps[stepId] = { status1Id: { stamp: .. , status: .., ..},  status2Id: { .. } }
certificates[certificateId] = { status1Id: { stamp: .. , status: .., ..},  status2Id: { .. } }
```
---

Therefore instead of doing here superfluous `exports.businessProcessNestedEntities.some`

We should just do regex match for status history event (same as e.g. we do for step keyPath: https://github.com/egovernment/eregistrations/blob/master/server/business-process-query/get-data.js#L135) which will confirm whether we deal with `statusHistory` record.

Also statusHistory records are not live, we do not need three-function handling via `validate`, `set`, `delete` which are dedicated for repetetive set/delete handling of same records.

It should be handled more straightforward way as e.g.

```javascript
var addStatusHistoryRecord = function (map, id, key, value) {
  if (!map[id]) map[id] = {};
  if (key === 'status') {
     if (value[0] !== '3') return;
     map[id].status = value;
  } else if (key === 'processor') {
     ...
  }
};

addStatusHistoryRecord(result._statusHistory.businessProcesses[businessProcessId],
statusHistoryId, statusHistoryKey, statusHistoryValue);
```